sdk/storage/azure-storage-files-shares/src/share_file_client.cpp

```diff
@@ -164,11 +164,9 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
     {
       protocolLayerOptions.FileContentDisposition = options.HttpHeaders.ContentDisposition;
     }
-    if (!options.HttpHeaders.ContentHash.Value.empty())
+    if (!options.HttpHeaders.ContentHash.Value.empty()
+        && options.HttpHeaders.ContentHash.Algorithm == HashAlgorithm::Md5)
     {
-      AZURE_ASSERT_MSG(
-          options.HttpHeaders.ContentHash.Algorithm == HashAlgorithm::Md5,
-          "This operation only supports MD5 content hash");
```

Comment:
Ditto on letting the server reject the request if the options are invalid.

Reply:
> Ditto on letting the server reject the request if the options are invalid.

Server won't reject it, server ignores this header.

Protocol layer code is auto-generated based on swagger (or something works like swagger). There's no way to add some headers that are not defined in swagger.